test/dtypes/test_fpx.py

```diff
@@ -69,61 +73,41 @@ def test_from_scaled_tc_fpx_compile(self, ebits, mbits, device):
-    @parametrize("ebits,mbits", _FPx_DTYPES)
-    @parametrize("leading_dims", [(4,), (2, 4)])
-    @parametrize("bias", [False, True])
-    def test_quant_llm_linear_weight(self, ebits, mbits, bias, leading_dims):
-        OC, IC = 256, 64
-        device = "cuda"
-
-        fp16_weight = torch.randn(OC, IC, device=device, dtype=torch.half)
-        fp16_bias = torch.randn(OC, device=device, dtype=torch.half) if bias else None
-
-        fpx_weight = QuantLlmLinearWeight.from_float(fp16_weight, ebits, mbits)
-
-        x = torch.randn(*leading_dims, IC, device=device, dtype=torch.half)
-        out = torch.nn.functional.linear(x, fpx_weight, fp16_bias)
-        assert out.shape == leading_dims + (OC,)
+        x = torch.randn(256, 64)
+        scale = choose_qparams_affine_fpx(x, ebits, mbits)
+        x = quantize_affine_fpx(x, scale, ebits, mbits)
+        layout_type = FpxTensorCoreLayoutType(ebits, mbits)
+        fpx_layout_tensor = FpxTensorCoreAQTLayout.from_plain(x, scale, None, layout_type).cuda()
+        assert fpx_layout_tensor.device.type == "cuda"
+        fpx_layout_tensor = fpx_layout_tensor.cpu()
+        assert fpx_layout_tensor.device.type == "cpu"
 
     @pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
+    @pytest.mark.skipif(not TORCH_VERSION_AT_LEAST_2_5, reason="quantization only works with torch.compile for 2.5+")
     @parametrize("ebits,mbits", _FPx_DTYPES)
     @parametrize("bias", [False, True])
-    def test_quant_llm_quantize(self, ebits, mbits, bias):
+    def test_fpx_weight_only(self, ebits, mbits, bias):
         N, OC, IC = 4, 256, 64
         device = "cuda"
 
-        linear = torch.nn.Linear(IC, OC, bias=bias, device=device)
+        # dtype must be float32
+        linear = torch.nn.Linear(IC, OC, bias=bias, device=device, dtype=torch.float32)
         fpx_linear = copy.deepcopy(linear)
-        quantize_(fpx_linear, quant_llm_fpx_weight_only(ebits, mbits))
+        quantize_(fpx_linear, fpx_weight_only(ebits, mbits))
 
-        x = torch.randn(N, IC, device=device, dtype=torch.half)
+        x = torch.randn(N, IC, device=device, dtype=torch.float32)
         expected = fpx_linear(x)
         actual = torch.compile(fpx_linear, fullgraph=True)(x)
-        torch.testing.assert_close(actual, expected)
-
-    @pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
-    def test_fp6_llm_quantize(self):
-        N, OC, IC = 4, 256, 64
-        device = "cuda"
-
-        linear = torch.nn.Linear(IC, OC, device=device)
-        fpx_linear = copy.deepcopy(linear)
-        quantize_(fpx_linear, fp6_llm_weight_only())
-
-        x = torch.randn(N, IC, device=device, dtype=torch.half)
-        expected = fpx_linear(x)
-        actual = torch.compile(fpx_linear, fullgraph=True)(x)
-        torch.testing.assert_close(actual, expected)
+        # somehow compile now changes the result a bit
+        torch.testing.assert_close(actual, expected, rtol=5e-4, atol=5e-4)
```

Comment:
Might be because the dtype is changed to FP32. The FPx kernel is using FP16 for computation.

Reply:
OK, I can revert this one, I thought it was required